substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/JavaMainWrapper.java

```diff
@@ -200,8 +209,12 @@ public static int run(int paramArgc, CCharPointerPointer paramArgv) {
      * @return maximum length of C chars that can be stored in the program argument part of the
      *         contiguous memory block without writing into the environment variables part.
      */
-    public static long getCRuntimeArgumentBlockLength() {
-        VMError.guarantee(argv.notEqual(WordFactory.zero()) && argc > 0, "Requires JavaMainWrapper.run(int, CCharPointerPointer) entry point!");
+    public static int getCRuntimeArgumentBlockLength() {
+        try {
+            checkArgumentBlockSupported();
+        } catch (UnsupportedOperationException e) {
```

Comment:
I don't like catching exceptions as normal control flow (and on Windows this is normal control flow). Better would be to make the `check` method return a boolean and in the callers either return `-1` or throw an exception.

Reply:
Will do.